codersdk/deployment.go

```diff
@@ -104,16 +122,145 @@ func (n FeatureName) AlwaysEnable() bool {
 		FeatureWorkspaceBatchActions:      true,
 		FeatureHighAvailability:           true,
 		FeatureCustomRoles:                true,
+		FeatureMultipleOrganizations:      true,
 	}[n]
 }
 
+// FeatureSet represents a grouping of features. Rather than manually
+// assigning features al-la-carte when making a license, a set can be specified.
+// Sets are dynamic in the sense a feature can be added to a set, granting the
+// feature to existing licenses out in the wild.
+// If features were granted al-la-carte, we would need to reissue the existing
+// old licenses to include the new feature.
+type FeatureSet string
+
+const (
+	FeatureSetNone       FeatureSet = ""
+	FeatureSetEnterprise FeatureSet = "enterprise"
+	FeatureSetPremium    FeatureSet = "premium"
+)
+
+func (set FeatureSet) Features() []FeatureName {
+	switch FeatureSet(strings.ToLower(string(set))) {
+	case FeatureSetEnterprise:
```

Comment:
A while ago Ammar did work to make it so we don't have to list new features in a bunch of places.

Instead of this, could we do the inverse where Premium simply detracts from the list instead? Seems easier to mentally model.

Reply:
If I understand correctly, you are saying we have 1 list, `All()` (it's FeatureNames() or something atm).

And we define `enterprise = All() - [multi-org]`, `premium = All()`. Rather than this explicit listing? 

If we define it that way, it feels more likely to accidentally include a feature in "enterprise", and then we'd have to revoke it later. This current method might be a bit of a nuisance to deal with, but it errors on the side of restrictive.

Now our unit tests do not use feature sets, they manually define features. So this might reveal something lacking in our unit tests that only running a real server will pickup :thinking:.

I'm still a bit split on the ideal way.

Reply:
I switch it :+1:. We should probably add some tests to assert we don't leak features into the wrong set, but not in this PR.

https://github.com/coder/coder/pull/13907/commits/7436159cc034c4019f82e185bd0d6716245cba35